src/net40/Radical.Windows.Presentation/Messaging/Handlers/ShowMessageBoxRequestHandler.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Windows;
+using Topics.Radical.ComponentModel.Messaging;
+using Topics.Radical.Messaging;
+using Topics.Radical.Validation;
+
+namespace Topics.Radical.Windows.Presentation.Messaging.Handlers
+{
+    /// <summary>
+    /// Handle the ShowMessageBoxRequest message
+    /// </summary>
+    public class ShowMessageBoxRequestHandler : AbstractMessageHandler<ShowMessageBoxRequest>, INeedSafeSubscription
+    {
+        private readonly IMessageBroker broker;
+
+        /// <summary>
+        /// ShowMessageBoxRequestHandler costructor
+        /// </summary>
+        /// <param name="broker"></param>
+        public ShowMessageBoxRequestHandler(IMessageBroker broker)
+        {
+            Ensure.That(broker).Named(() => broker).IsNotNull();
```

Comment:
do not use the labda, it's too slow. Replace with `.Named("broker")` or better yet `.Named(nameof(broker))`.